projects/sound-open-firmware/build.sh

```diff
@@ -19,12 +19,30 @@
 #
 ################################################################################
 
-cd $SRC/sof/tools/oss-fuzz
-cp corpus/* $OUT/
-rm -rf build_oss_fuzz
-mkdir -p build_oss_fuzz
-cd build_oss_fuzz
+# Environment: Zephyr has its own toolchain selection mechanism, and
+# the oss-fuzz environment generates warnings and confuses things when
+# building some of the (un-fuzzed) build-time tools that aren't quite
+# clang-compatible.
+export ZEPHYR_TOOLCHAIN_VARIANT=llvm
+unset CC
+unset CCC
+unset CXX
+unset CFLAGS
+unset CXXFLAGS
 
-export VERBOSE=1
-cmake -DCMAKE_INSTALL_PREFIX=install -DCMAKE_LINKER=$CXX -DCMAKE_C_LINK_EXECUTABLE="<CMAKE_LINKER> <FLAGS> <CMAKE_C_LINK_FLAGS> <LINK_FLAGS> <OBJECTS> -o <TARGET> <LINK_LIBRARIES>" ..
-make install -j $(nproc)
+BASE_CFG=`cat <<EOF
```

Comment:
This seems unnecessary, bash supports multilines strings?

```suggestion
BASE_CFG='
-DCONFIG_ASSERT=y
...
```
```

Reply:
The recommended and shellcheck-compliant way to do this is using a bash array:

```bash
BASE_CFG=(
-DCONFIG_ASSERT=y
-DCONFIG_SYS_HEAP_BIG_ONLY=y
...
)
declare -p BASE_CFG
west build -p -b native_posix ./app -- "${BASE_CFG[@]}"
```
Shellcheck saves lives; use it always. It's also a great... teacher.

You can't go wrong with arrays, they're _always_ safe.

http://mywiki.wooledge.org/BashGuide/Arrays
http://mywiki.wooledge.org/Quotes

Reply:
resolved